nf_core/modules.py

```diff
@@ -231,3 +240,259 @@ def has_valid_pipeline(self):
         if not os.path.exists(main_nf) and not os.path.exists(nf_config):
             log.error("Could not find a main.nf or nextfow.config file in: {}".format(self.pipeline_dir))
             return False
+
+
+class ModulesTestMetaBuilder(object):
+    def __init__(
+        self,
+        module_name,
+        run_tests=False,
+        test_yml_output_path=None,
+        force_overwrite=False,
+        no_prompts=False,
+    ):
+        self.module_name = module_name
+        self.run_tests = run_tests
+        self.test_yml_output_path = test_yml_output_path
+        self.force_overwrite = force_overwrite
+        self.no_prompts = no_prompts
+        self.module_dir = os.path.join("software", *module_name.split("/"))
+        self.module_test_main = os.path.join("tests", "software", *module_name.split("/"), "main.nf")
+        self.entry_points = []
+        self.tests = []
+
+    def run(self):
+        """ Run build steps """
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self.check_inputs()
+        self.scrape_workflow_entry_points()
+        self.build_all_tests()
+        self.print_test_yml()
+
+    def check_inputs(self):
+        """ Do more complex checks about supplied flags. """
+        # First, sanity check that the module directory exists
+        if not os.path.isdir(self.module_dir):
+            raise UserWarning(f"Cannot find directory '{self.module_dir}'. Should be TOOL/SUBTOOL or TOOL")
+        if not os.path.exists(self.module_test_main):
+            raise UserWarning(f"Cannot find module test workflow '{self.module_dir}/main.nf'")
```

Comment:
I think that's checking if `tests/software/tool/main.nf` exists, but warns about `software/tool/main.nf`

Reply:
Made this mistake several times over now 😞

Reply:
Fixed 👍🏻